src/drm.ts

```diff
@@ -0,0 +1,343 @@
+   * string. So use whatever solution suits you best to handle Base64 in React Native.
+   *
+   * @param message - Base64 encoded message data.
+   * @param assetId - Asset ID.
+   * @returns The processed Base64 encoded sync message.
+   */
+  prepareSyncMessage?: (syncMessage: string, assetId: string) => string;
+  /**
+   * A block to prepare the URI (without the skd://) from the HLS manifest before passing it to the
+   * system.
+   *
+   * @param licenseServerUrl - License server URL string.
+   * @returns The processed license server URL string.
+   */
+  prepareLicenseServerUrl?: (licenseServerUrl: string) => string;
+  /**
+   * A block to prepare the `contentId`, which is sent to the FairPlay Streaming license server as
+   * request body, and which is used to build the SPC data. As many DRM providers expect different,
+   * vendor-specific messages, this can be done using this user-defined block. The parameter is the
+   * skd:// URI extracted from the HLS manifest (m3u8) and the return value should be the contentID
+   * as string.
+   *
+   * @param contentId - Extracted content id string.
+   * @returns The processed contentId.
+   */
+  prepareContentId?: (contentId: string) => string;
+  /**
+   * A block to provide the license for the given `assetId`.
+   *
+   * This should only be used when the license is stored locally. If the block returns nil we will
+   * try to fetch the license using the available information in the same `FairplayConfig`.
+   *
+   * If no `certificateUrl` is present at this point we will emit a `SourceErrorEvent`.
+   *
+   * Use this block for the following use-cases:
+   *
+   * - License caching during playback or for future playback sessions.
+   *
+   * If the provided license is expired, the playback will fail. License duration handling has to be
+   * handled by the application.
+   *
+   * Note that this block return value should be a Base64 string. So use whatever solution suits you
+   * best to handle Base64 in React Native.
+   *
+   * @param assetId - Provided `assetId` value.
+   * @returns The persisted Base64 encoded license data.
+   */
+  provideLicenseData?: (assetId: string) => string | null;
+  /**
+   * A block to enable custom persisting of license data for the given `assetId`.
+   *
+   * Use this block for the following use-cases:
+   *
+   * - To store the license data locally for future playback sessions.
+   * - To update the license during a playback session if the license requires updating.
+   *
+   * In both use-cases, it’s required that the license is persistable on the device.
+   * Playback will fail otherwise.
+   *
+   * Note that the `licenseData` is provided as a Base64 string. So use whatever solution suits you
+   * best to handle Base64 in React Native.
+   *
+   * @param assetId - Provided `assetId` value.
+   * @param licenseData - License data as a Base64 encoded string.
+   */
+  persistLicenseData?: (assetId: string, licenseData: string) => void;
```

Comment:
nit: These two properties could be omitted for now. These are only for advanced offline license caching scenarios.

Reply:
Done 693064232f4465a2facd25e3342c2f8606890f8b